Editor/AGS.Editor/Panes/FreePanControl.cs

```diff
@@ -0,0 +1,96 @@
+
+namespace AGS.Editor
+{
+    public class FreePanControl : BufferedPanel
+    {
+        private const int WS_EX_COMPOSITED = 0x02000000;
+        private int _lastX, _lastY; //used to prevent useless mouse move events
+        private Point _panGrabPoint = Point.Empty;
+        private Cursor _panCursor = Cursors.Hand; // We need a better hand cursor, using the "link" hand for now
+        public event EventHandler PanGrabbed;
+        public event EventHandler PanReleased;
+        public bool IsPanning { get; private set; }
+
+        // Enabling Composite makes scrolling smoother
+        protected override CreateParams CreateParams
+        {
+            get
+            {
+                CreateParams cp = base.CreateParams;
+                cp.ExStyle |= WS_EX_COMPOSITED;
+                return cp;
+            }
+        }
+
+        protected virtual void OnPanGrab(EventArgs e)
+        {
+            PanGrabbed?.Invoke(this, EventArgs.Empty);
+        }
+
+        protected virtual void OnPanRelease(EventArgs e)
+        {
+            PanReleased?.Invoke(this, EventArgs.Empty);
+        }
+
+        protected override void OnMouseDown(MouseEventArgs e)
+        {
+            // defines the shortcut for click pan Control+Shift+Left Click Hold or
+            // Middle Click Hold without any modifier.
+            if ( (e.Button == MouseButtons.Left   && ModifierKeys == (Keys.Control|Keys.Shift)) ||
+                 (e.Button == MouseButtons.Middle && ModifierKeys == Keys.None) )
+            {
+                OnPanGrab(e);
+                _panGrabPoint = e.Location;
+                Cursor = _panCursor;
+                IsPanning = true;
+            }
+
+            base.OnMouseDown(e);
+        }
+
+        protected override void OnMouseUp(MouseEventArgs e)
+        {
+            if (_panGrabPoint != Point.Empty)
+            {
+                _panGrabPoint = Point.Empty;
+                Cursor = Cursors.Default;
+                IsPanning = false;
+                OnPanRelease(e);
+            }
+
+            base.OnMouseUp(e);
+        }
+
+        protected override void OnMouseMove(MouseEventArgs e)
+        {
+            if ((e.X == _lastX) && (e.Y == _lastY))
+            {
+                return;
+            }
+
+            _lastX = e.X;
+            _lastY = e.Y;
+
+            if (_panGrabPoint != Point.Empty)
+            {
+                Point scrollPosition = AutoScrollPosition;
+                scrollPosition.X = _panGrabPoint.X - e.X - scrollPosition.X;
+                scrollPosition.Y = _panGrabPoint.Y - e.Y - scrollPosition.Y;
+                AutoScrollPosition = scrollPosition;
+                Refresh(); // this prevents the room image to look garbled when panning
+                _panGrabPoint = e.Location; // <- without this the pan doesn't work
+            }
+
+            base.OnMouseMove(e);
+        }
+        public Cursor GetPanCursor() { return _panCursor; }
```

Comment:
You can just get rid of the field `_panCursor` and instead do this

```
public Cursor PanCursor { get; private set; }
```

Rule of thumb; Try to use fields/properties to read values of the class, and save methods for doing calculations on the class.

Put the property with the rest of the properties in the class

Reply:
I did it! :grinning: